src/main/scripts/src/app/shared/search-box/search-box.component.ts

```diff
@@ -224,14 +377,66 @@ export class SearchBoxComponent implements OnInit {
   }
 
   private addQueryItem(queryItem: QueryItem) {
-    this.queryItems.push(queryItem);
-    this.currentQueryItem = new FulltextQueryItem('');
-    this.text = '';
-    this.hideSuggestions();
+    if (queryItem.type == QueryItemType.Condition) {
+      if (this.selectedQueryItem >= 0 && this.queryItems[this.selectedQueryItem].type == QueryItemType.Attribute) {
+        (this.queryItems[this.selectedQueryItem] as AttributeQueryItem).condition = queryItem.text;
+        const conditionElement = this.findConditionElement(this.selectedQueryItem);
+        if (conditionElement) {
+          setTimeout(() => {
+            conditionElement.nativeElement.focus()
+          }, 200);
+        }
+        this.selectedSuggestion = -1;
+      }
+    } else {
+      if (queryItem.type === QueryItemType.Attribute) {
+        const collectionItem = (queryItem as AttributeQueryItem).toCollectionQueryItem();
+        if (this.queryItems.length === 0 || !this.isOneOfLastItems(collectionItem)) {
+          this.queryItems.push(collectionItem);
+        }
+        this.shouldFocusCondition = true;
+      }
+      this.queryItems.push(queryItem);
+      this.text = '';
+    }
+    this.clearSuggestions();
+  }
+
+  private findConditionElement(id: number): ElementRef {
+    const filtered = this.conditions.filter(element => element.nativeElement.id == id);
+    return filtered.length == 1 ? filtered[0] : undefined;
+  }
+
+  private isOneOfLastItems(queryItem: CollectionQueryItem): boolean {
+    if (this.queryItems.length > 0) {
+      for (let i = this.queryItems.length - 1; i >= 0; i--) {
+        const item = this.queryItems[i];
+        if (item.type == QueryItemType.Collection) {
+          return item.value === queryItem.value;
+        }
+      }
+    }
+    return false;
   }
 
   public removeHtmlComments(html: HTMLElement): string {
     return HtmlModifier.removeHtmlComments(html);
   }
 
+  private isFullTextQueryPresented(): boolean {
+    return this.queryItems.some(item => item.type === QueryItemType.Fulltext);
+  }
+
+  public isCollectionItem(queryItem: QueryItem): boolean {
+    return queryItem.type === QueryItemType.Collection;
+  }
+
+  public hex2rgba(hex: string, opacity: number): string {
```

Comment:
You do not need to deal with the opacity at the moment. I have developed a solution for this which will be added in my next pull request.